src/components/flat-table/flat-table.pw.tsx

```diff
@@ -0,0 +1,3619 @@
+        ).toHaveText(record);
+      });
+    }
+  );
+
+  ([
+    ["large", 700, 345],
+    ["small", 700, 240],
+  ] as [string, number, number][]).forEach(
+    ([size, tableWidth, tableHeight]) => {
+      test(`should render with pager and sticky header in ${size} viewport`, async ({
+        mount,
+        page,
+      }) => {
+        await page.setViewportSize({ width: tableWidth, height: tableHeight });
+        await mount(<FlatTablePagerStickyHeaderComponent />);
+
+        await flatTablePageSizeSelect(page).click();
+
+        if (size === "large") {
+          const selectListPos = flatTablePageSelectListPosition(page);
+          await expect(selectListPos).toHaveAttribute(
+            "data-floating-placement",
+            "bottom"
+          );
+          await expect(selectListPos).toBeInViewport();
+        } else {
+          const selectListPos = flatTablePageSelectListPosition(page);
+          await expect(selectListPos).toHaveAttribute(
+            "data-floating-placement",
+            "top"
+          );
+          await expect(selectListPos).toBeInViewport();
+        }
+      });
+    }
+  );
+
+  ([
+    [1, 0],
+    [5, 1],
+  ] as const).forEach(([numberOfItems, option]) => {
+    test(`should render with ${numberOfItems} items when selected with the mouse`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<FlatTablePagerStickyHeaderComponent />);
+
+      const tableBodyRows = flatTableBodyRows(page);
+      await expect(tableBodyRows).toHaveCount(5);
+      await expect(flatTableBodyRows(page).first()).toBeInViewport();
+      await pageSelectInput(page).click();
+      await flatTablePageSelectListPosition(page)
+        .locator("li")
+        .nth(option)
+        .click();
+      await expect(tableBodyRows).toHaveCount(numberOfItems);
+      await expect(flatTableBodyRows(page).first()).toBeInViewport();
+      await expect(pageSelectInput(page)).toHaveValue(numberOfItems.toString());
+    });
+  });
+
+  ([
+    [1, 0],
+    [5, 1],
+  ] as const).forEach(([numberOfItems, option]) => {
+    test(`should open Show Items selector with the Spacebar and select ${numberOfItems} items per page`, async ({
+      mount,
+      page,
+    }) => {
+      await mount(<FlatTablePagerStickyHeaderComponent />);
+
+      const tableBodyRows = flatTableBodyRows(page);
+      await expect(tableBodyRows).toHaveCount(5);
+      await expect(flatTableBodyRows(page).first()).toBeInViewport();
+      await pageSelectInput(page).focus();
+      await page.keyboard.press("Space");
+      await flatTablePageSelectListPosition(page)
+        .locator("li")
+        .nth(option)
+        .click();
+      await expect(tableBodyRows).toHaveCount(numberOfItems);
+      await expect(flatTableBodyRows(page).first()).toBeInViewport();
+      await expect(pageSelectInput(page)).toHaveValue(numberOfItems.toString());
+    });
+  });
+
+  // SKIPPED - Enter key does not open Pager dropdowns.
```

Comment:
Can you please log a bug for this against the Pager component? It seems it requires two presses of the Enter key to open this Select in Pager which is not correct. It should open with one here.